pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1309,3 +1332,79 @@ func initializeAnnotations(cd *hivev1.ClusterDeployment) {
 		cd.Annotations = map[string]string{}
 	}
 }
+
+// managePullSecrets() takes pull secretes in to account and returns the merged pull request
+func (r *ReconcileClusterDeployment) managePullSecrets(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (string, error) {
+
+	// For code readability let's call the pull secret in cluster deployment config as local pull secret
+	localPullSecret, err := controllerutils.LoadSecretData(r.Client, cd.Spec.PullSecret.Name, cd.Namespace, corev1.DockerConfigJsonKey)
```

Comment:
Need to check that the pull secret name was supplied, since it is optional. We should not be warning about being unable to load the secret when it was omitted by the user.